File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    
    stack=[]
    top=-1
    for col in moves:
        row=0
        while row<len(board):
            if board[row][col-1]!=0:
                catch=board[row][col-1]
                board[row][col-1]=0
                if top==catch:
                    stack.pop()
                    if len(stack)==0: top=-1
                    else: top=stack[-1]
                    answer+=2
                else:
                    stack.append(catch)
                    top=stack[-1]
                break
            else:
                row+=1
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
         int answer = 0;
        int len=board.length;
        Stack<Integer>[] st=new Stack[len];
        
        for(int i=0;i<len;i++) st[i]=new Stack<Integer>();
      
       
        
        for(int row=len-1;row>=0;row--){
            for(int col=0;col<len;col++){
                if(board[row][col]!=0) st[col].push(board[row][col]);
            }
        }
        Stack<Integer> result=new Stack<>();
        int before=-1;
        for(int m:moves){
            if(result.size()>0){
                before=result.peek();
            }
            if(!st[m-1].isEmpty()){
            int now=st[m-1].pop();
            result.push(now);
        
            if(before==now){
                result.pop();
                result.pop();
                before=-1;
                answer++;
                answer++;
            }
            }
        }
       
        return answer;
    }
}